### PR TITLE
Add oslog usage on macOS

### PR DESCRIPTION
**Pull request by rokuz**
Xcode 15 is coloring the console automatically if we use oslog.

Снимок экрана 2023-09-22 в 10 08 53

### CODE REVIEW

File: minilog.cpp [hunk 1]
```diff
@@ -96,6 +100,21 @@ struct ThreadLogContext
 	bool hasLogsOnThisLevel[kMaxProcsNesting] = { false };
 };
 
+#if OS_MACOS
+static os_log_type_t logLevelToOsLogType(minilog::eLogLevel level) 
+{
+	switch (level) 
+	{
+		case minilog::eLogLevel::Paranoid: return OS_LOG_TYPE_INFO;
```
Comment (corporateshark) [inline minilog.cpp:108]: `Paranoid` is more verbose than `Debug`. Is it the same semantics with `OG_LOG_TYPE_INFO` and `OG_LOG_TYPE_DEBUG`? Which one is more verbose?

File: minilog.cpp [hunk 2]
```diff
@@ -353,10 +385,13 @@ static void printMessageToConsole(minilog::eLogLevel level, const char* msg, con
 		} else {
 			printf(FORMATSTR_NO_THREAD, msg);
 		}
+#endif
```
Comment (corporateshark) [inline minilog.cpp]: `// OS_MACOS`

**Review by corporateshark — CHANGES_REQUESTED** [review]

**Comment by corporateshark** [comment]
Please update that comment here https://github.com/corporateshark/minilog/blob/main/minilog.h#L31

File: minilog.cpp (hunk 1 above)
Comment (rokuz) [reply minilog.cpp:108]: I'm not sure Apple has concept of verbosity in oslog, it's more filter. You can enable and disable any category. Apple's classification:
```
 * @constant OS_LOG_TYPE_DEFAULT
 * Equivalent type for "os_log()" messages, i.e., default messages that are
 * always captured to memory or disk.
 *
 * @constant OS_LOG_TYPE_INFO
 * Equivalent type for "os_log_info()" messages, i.e., additional informational
 * messages.
 *
 * @constant OS_LOG_TYPE_DEBUG
 * Equivalent type for "os_log_debug()" messages, i.e., debug messages.
 *
 * @constant OS_LOG_TYPE_ERROR
 * Equivalent type for "os_log_error()" messages, i.e., messages indicating
 * error conditions.
 *
 * @constant OS_LOG_TYPE_FAULT
 * Equivalent type for "os_log_fault()" messages, i.e., messages indicating
 * that an unexpected condition occurred that likely indicates the presence of a
 * bug.
```

Default - base level, equal to minilog::eLogLevel::Log
Paranoid - the closest to OS_LOG_TYPE_INFO, so we output additional information not typically required.

File: minilog.cpp [hunk 3]
```diff
@@ -344,19 +365,39 @@ static void printMessageToConsole(minilog::eLogLevel level, const char* msg, con
...
+		if (usePrintf) {
+			if (config.threadNames) {
+				if (ctx->threadName) {
+					printf(FORMATSTR_THREAD_NAME, ctx->threadName, msg);
+				} else {
+					printf(FORMATSTR_THREAD_ID, (unsigned long long)ctx->threadId, msg);
+				}
 			} else {
-				printf(FORMATSTR_THREAD_ID, (unsigned long long)ctx->threadId, msg);
+				printf(FORMATSTR_NO_THREAD, msg);
```
Comment (corporateshark) [inline minilog.cpp:392]: This should not be changed.

**Review by corporateshark — APPROVED** [review]

**Pull request merged**